### PR TITLE
refactor websocket key hashing

### PR DESCRIPTION
**Pull request by robjtede**
## PR Type

Refactor

## PR Checklist
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

## Overview
WebSocket key hashing does not allocate String.

### CODE REVIEW

File: actix-http/examples/ws.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+};
+
+use actix_codec::Encoder;
+use actix_http::{error::Error, ws, HttpService, Request, Response};
+use actix_rt::time::{interval, Interval};
+use actix_server::Server;
+use bytes::{Bytes, BytesMut};
+use bytestring::ByteString;
+use futures_core::{ready, Stream};
+use http::StatusCode;
```
Comment (ibraheemdev) [inline actix-http/examples/ws.rs]: You should probably fix this unused import.

**Review by ibraheemdev — APPROVED** [review]
:+1:

File: actix-http/examples/ws.rs (hunk 1 above)
Comment (robjtede) [inline actix-http/examples/ws.rs]:
```suggestion
```

**Review by fakeshadow — APPROVED** [review]
LGTM

**Pull request merged**